File: lib/cfr/builtins/libs/dart/ui/fontWeight.dart
import 'dart:core';
import 'dart:ui';

import 'package:hydro_sdk/cfr/runtimeSupport.dart';

class VMManagedFontWeight extends VMManagedBox<FontWeight> {
  VMManagedFontWeight(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {
    table['index'] = vmObject.index;
    table['toString'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.toString(),
      ];
    });
  }

  final HydroTable table;

  final HydroState hydroState;

  final FontWeight vmObject;
}

void loadFontWeight(
    {required HydroState hydroState, required HydroTable table}) {
  table['fontWeightW100'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<FontWeight>(
          object: FontWeight.w100, hydroState: hydroState, table: HydroTable())
    ];
  });
  table['fontWeightW200'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<FontWeight>(
          object: FontWeight.w200, hydroState: hydroState, table: HydroTable())
    ];
  });
  table['fontWeightW300'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<FontWeight>(
          object: FontWeight.w300, hydroState: hydroState, table: HydroTable())
    ];
  });
  table['fontWeightW400'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<FontWeight>(
          object: FontWeight.w400, hydroState: hydroState, table: HydroTable())
    ];
  });
  table['fontWeightW500'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<FontWeight>(
          object: FontWeight.w500, hydroState: hydroState, table: HydroTable())
    ];
  });
  table['fontWeightW600'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<FontWeight>(
          object: FontWeight.w600, hydroState: hydroState, table: HydroTable())
    ];
  });
  table['fontWeightW700'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<FontWeight>(
          object: FontWeight.w700, hydroState: hydroState, table: HydroTable())
    ];
  });
  table['fontWeightW800'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<FontWeight>(
          object: FontWeight.w800, hydroState: hydroState, table: HydroTable())
    ];
  });
  table['fontWeightW900'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<FontWeight>(
          object: FontWeight.w900, hydroState: hydroState, table: HydroTable())
    ];
  });
  table['fontWeightLerp'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    final returnValue = FontWeight.lerp(
        maybeUnBoxAndBuildArgument<FontWeight?, dynamic>(luaCallerArguments[1],
            parentState: hydroState),
        maybeUnBoxAndBuildArgument<FontWeight?, dynamic>(luaCallerArguments[2],
            parentState: hydroState),
        luaCallerArguments[3]?.toDouble());
    if (returnValue != null) {
      return [
        maybeBoxObject<FontWeight?>(
            object: returnValue, hydroState: hydroState, table: HydroTable()),
      ];
    }
    return [];
  });
  registerBoxer<FontWeight>(boxer: (
      {required FontWeight vmObject,
      required HydroState hydroState,
      required HydroTable table}) {
    return VMManagedFontWeight(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}

File: runtime/dart/ui/fontWeight.ts
import { List } from "../core/list";
declare const dart: {
    ui: {
        fontWeightLerp: (
            a: IFontWeight | undefined,
            b: IFontWeight | undefined,
            t: number
        ) => IFontWeight | undefined;
        fontWeightW100: (this: void) => IFontWeight;
        fontWeightW200: (this: void) => IFontWeight;
        fontWeightW300: (this: void) => IFontWeight;
        fontWeightW400: (this: void) => IFontWeight;
        fontWeightW500: (this: void) => IFontWeight;
        fontWeightW600: (this: void) => IFontWeight;
        fontWeightW700: (this: void) => IFontWeight;
        fontWeightW800: (this: void) => IFontWeight;
        fontWeightW900: (this: void) => IFontWeight;
    };
};
export interface IFontWeight {
    index: number;
    toString: () => string;
}
export class FontWeight {
    public static w100 = dart.ui.fontWeightW100();
    public static w200 = dart.ui.fontWeightW200();
    public static w300 = dart.ui.fontWeightW300();
    public static w400 = dart.ui.fontWeightW400();
    public static w500 = dart.ui.fontWeightW500();
    public static w600 = dart.ui.fontWeightW600();
    public static w700 = dart.ui.fontWeightW700();
    public static w800 = dart.ui.fontWeightW800();
    public static w900 = dart.ui.fontWeightW900();
    public static normal = FontWeight.w400;
    public static bold = FontWeight.w700;
    public static values = List.fromArray([
        FontWeight.w100,
        FontWeight.w200,
        FontWeight.w300,
        FontWeight.w400,
        FontWeight.w500,
        FontWeight.w600,
        FontWeight.w700,
        FontWeight.w800,
        FontWeight.w900,
    ]);
    public readonly index: number = undefined as any;
    public static lerp(
        a: IFontWeight | undefined,
        b: IFontWeight | undefined,
        t: number
    ): IFontWeight | undefined {
        return dart.ui.fontWeightLerp(a, b, t);
    }
    private readonly _dart_toString: () => string = undefined as any;
    public toString(): string {
        return this._dart_toString();
    }
}
